# 기본 데이터 스크래핑

In [1]:
import requests
from bs4 import BeautifulSoup

url = "https://www.musinsa.com/ranking/best?period=month&age=ALL&mainCategory=002&subCategory=&leafCategory=&price=&golf=false&kids=false&newProduct=false&exclusive=false&discount=false&soldOut=false&page=1&viewType=small&priceMin=&priceMax="

user_agent = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36"

headers = {"User-Agent" : user_agent}
response = requests.get(url, headers)
response.raise_for_status()

soup = BeautifulSoup(response.text, 'lxml')

In [46]:
def specific_info(link_list):
    headers = {'User-Agent':'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/101.0.4951.67 Safari/537.36'}
    
    sales_html = html_2.find_all("strong", {"id":"sales_1y_qty"})
    
    sales = sales_html[0].get_text()
    sales_list.append(sales)

In [10]:
# 브랜드
brands = soup.select('form#goodsRankForm div.list-box.box div.article_info p.item_title a')

for brand in brands :

    print(brand.text)

드로우핏
스파오
내셔널지오그래픽
토피
노스페이스
쿠어
라퍼지스토어
스파오
파르티멘토
스파오
라퍼지스토어
코드그라피
예일
도프제이슨
예일
알파 인더스트리
비슬로우
무신사 스탠다드
코드그라피
내셔널지오그래픽
무신사 스탠다드
라퍼지스토어
라퍼지스토어
커버낫
라퍼지스토어
라퍼지스토어
내셔널지오그래픽
노이어
디스커버리 익스페디션
도프제이슨
무신사 스탠다드
무신사 스탠다드
내셔널지오그래픽
쿠어
무신사 스탠다드
라퍼지스토어
커버낫
커버낫
무신사 스탠다드
쿠어
무신사 스탠다드
후아유
드로우핏
커버낫
쿠어
예일
후아유
노스페이스
커버낫
키뮤어
라퍼지스토어
도프제이슨
무신사 스탠다드
내셔널지오그래픽
제멋
내셔널지오그래픽
메종미네드
디스커버리 익스페디션
파르티멘토
코드그라피
코드그라피
드로우핏
노스페이스
게스
코드그라피
인사일런스
케이투
후아유
무신사 스탠다드
예일
키뮤어
커버낫
무신사 스탠다드
라퍼지스토어
커버낫
라퍼지스토어
후아유
프리즘웍스
무신사 스탠다드
엠엘비
비슬로우
인사일런스
캘빈클라인 진
스파오
토피
파르티멘토
쿠어
디스이즈네버댓
노스페이스
라퍼지 포 우먼


In [17]:
import selenium
from selenium import webdriver
import time
driver = webdriver.Chrome()
for i in range(90) : 
    driver.get("https://www.musinsa.com/ranking/best?period=month&age=ALL&mainCategory=002&subCategory=&leafCategory=&price=&golf=false&kids=false&newProduct=false&exclusive=false&discount=false&soldOut=false&page=1&viewType=small&priceMin=&priceMax=")
    items = driver.find_elements_by_css_selector("img.lazyload.lazy")
    items[i].click()
    time.sleep(1)
    sum_sale = driver.find_element_by_id("sales_1y_qty")
    sst = sum_sale.text

    if '만' in sst :
        if '.' in sst : 
            sst = sst.replace('.','')
            sst += '000'
        else :
            sst += '0000'
            
        sst = sst.replace('만 개 이상','')
        
    else :
        if '.' in sst : 
            sst = sst.replace('.','')
            sst += '00'
        else :
            sst += '000'
            
        sst = sst.replace('천 개 이상','')
        
    print(sst)

8600
41000


KeyboardInterrupt: 

ok


In [66]:
# 제품명
names = soup.select('form#goodsRankForm div.list-box.box div.article_info p.list_info a')

for name in names :
    print(name['title'])

프리미어 퓨어 캐시미어 히든 싱글 코트 [BLACK]
베이직 푸퍼_SPJPC4TC11
N224MDW920 헤론 U넥 경량 다운 점퍼 CARBON BLACK
2WAY 스웻 후드 집업 (MELANGE GREY)
NJ1DN75A 남성 1996 에코 눕시 자켓
MTR 발마칸 코트 (블랙)
[리뉴얼 ver.][3M 신슐레이트]오리지널 M-1965 피쉬테일 파카_Original Khaki
베이직 퍼플리스 집업_SPFZC4TU01
[22F/W 리뉴얼] 오버사이즈 멜톤 더플 롱 코트 네이비
파스텔 푸퍼_SPJPC4TG01
버핑레더 오버핏 블루종_Black
[내피분리형] US-ARMY 밀리터리 피쉬테일 후드 롱파카_카키
REVERSIBLE WARM UP QUILTING JACKET BLACK / IVORY
오버핏 비건레더 싱글 자켓
(22FW) WARM+ UP QUILTING JACKET BLACK
MA-1 레귤러 Black
22FW OVERSIZED CASHMERE BALMACAAN COAT BLACK
캐시미어 블렌드 오버사이즈 싱글 코트 [블랙]
[DIVO] RDS 유틸리티 덕다운 푸파 숏패딩_블랙
N224MFJ910 남성 코스토니 플리스 뽀글이 후드 풀집업 SUGAR SWIZZLE IVORY
캐시미어 블렌드 오버사이즈 발마칸 코트 [블랙]
[리뉴얼 ver.][3M 신슐레이트] M-1965 피쉬테일 숏 파카_Vintage Black
울리치 발마칸 오버 코트_Black
리버시블 쉐르파 집업 자켓 아이보리
[무봉제]덕다운 심리스 미니멀 푸퍼 숏패딩_Black
[리뉴얼 ver.][3M 신슐레이트]오리지널 M-1965 피쉬테일 파카_Vintage Black
N224WDW920 여성 헤론 U넥 경량 덕다운 점퍼 L BEIGE
캐시미어 더블 브레스티드 코트 (블랙)
픽토르 U 구스다운 경량패딩 (BLACK)
솔리드 무톤 자켓
릴렉스드 베이식 블레이저 [블랙]
오버사이즈 MA-1 보머 재킷 [블랙]
N224MFJ910 남성 코스토니 플리스 뽀글이 후드 풀집업 CARBON BLACK
M

In [67]:
# 가격대
prices = soup.select('div.right_area.page-contents-best form#goodsRankForm div.list-box.box div.article_info p.price')

for price in prices :
    prices = price.text.split()
    if len(prices) == 2:
        print("할인전 가격: ", prices[0])
        print("할인후 가격: ", prices[1])
    else:
        print("할인없는 제품", prices[0])
    print('*'*20)

할인전 가격:  278,000원
할인후 가격:  219,000원
********************
할인없는 제품 69,900원
********************
할인없는 제품 149,000원
********************
할인전 가격:  45,000원
할인후 가격:  42,000원
********************
할인없는 제품 339,000원
********************
할인전 가격:  329,000원
할인후 가격:  296,100원
********************
할인전 가격:  138,000원
할인후 가격:  99,900원
********************
할인없는 제품 29,900원
********************
할인전 가격:  155,000원
할인후 가격:  124,000원
********************
할인없는 제품 69,900원
********************
할인전 가격:  94,000원
할인후 가격:  69,900원
********************
할인전 가격:  139,000원
할인후 가격:  98,900원
********************
할인전 가격:  129,000원
할인후 가격:  103,200원
********************
할인전 가격:  129,000원
할인후 가격:  89,000원
********************
할인전 가격:  99,000원
할인후 가격:  74,250원
********************
할인없는 제품 279,000원
********************
할인전 가격:  299,000원
할인후 가격:  254,150원
********************
할인없는 제품 159,900원
********************
할인전 가격:  124,000원
할인후 가격:  99,900원
********************
할인전 가격:  179,000원
할인후 가격:  159,000원
********************
할인없는 제

In [82]:
# 다 넣은 곳에서 쿠폰 있는거 없는거 찾기
all_items = soup.select('div.right_area.page-contents-best form#goodsRankForm div.list-box.box .li_box')
brands = soup.select('form#goodsRankForm div.list-box.box div.article_info p.item_title a')

for i, brand in zip(all_items, brands) :
    a = i.find('p', { 'class' : 'mu-icon mu-icon-coupon'})
    if a == None :
        print(brand.text, "쿠폰: x")
    else :
        print (brand.text, "쿠폰: o")

드로우핏 쿠폰: x
스파오 쿠폰: o
내셔널지오그래픽 쿠폰: x
토피 쿠폰: x
노스페이스 쿠폰: x
쿠어 쿠폰: x
라퍼지스토어 쿠폰: o
스파오 쿠폰: o
파르티멘토 쿠폰: o
스파오 쿠폰: o
라퍼지스토어 쿠폰: o
코드그라피 쿠폰: x
예일 쿠폰: o
도프제이슨 쿠폰: x
예일 쿠폰: o
알파 인더스트리 쿠폰: x
비슬로우 쿠폰: x
무신사 스탠다드 쿠폰: x
코드그라피 쿠폰: x
내셔널지오그래픽 쿠폰: x
무신사 스탠다드 쿠폰: x
라퍼지스토어 쿠폰: o
라퍼지스토어 쿠폰: o
커버낫 쿠폰: x
라퍼지스토어 쿠폰: o
라퍼지스토어 쿠폰: o
내셔널지오그래픽 쿠폰: x
노이어 쿠폰: x
디스커버리 익스페디션 쿠폰: x
도프제이슨 쿠폰: x
무신사 스탠다드 쿠폰: x
무신사 스탠다드 쿠폰: x
내셔널지오그래픽 쿠폰: x
쿠어 쿠폰: x
무신사 스탠다드 쿠폰: x
라퍼지스토어 쿠폰: o
커버낫 쿠폰: x
커버낫 쿠폰: x
무신사 스탠다드 쿠폰: o
쿠어 쿠폰: x
무신사 스탠다드 쿠폰: x
후아유 쿠폰: x
드로우핏 쿠폰: x
커버낫 쿠폰: x
쿠어 쿠폰: x
예일 쿠폰: o
후아유 쿠폰: x
노스페이스 쿠폰: x
커버낫 쿠폰: x
키뮤어 쿠폰: x
라퍼지스토어 쿠폰: o
도프제이슨 쿠폰: x
무신사 스탠다드 쿠폰: o
내셔널지오그래픽 쿠폰: x
제멋 쿠폰: x
내셔널지오그래픽 쿠폰: x
메종미네드 쿠폰: x
디스커버리 익스페디션 쿠폰: x
파르티멘토 쿠폰: o
코드그라피 쿠폰: o
코드그라피 쿠폰: o
드로우핏 쿠폰: x
노스페이스 쿠폰: x
게스 쿠폰: x
코드그라피 쿠폰: o
인사일런스 쿠폰: x
케이투 쿠폰: x
후아유 쿠폰: x
무신사 스탠다드 쿠폰: x
예일 쿠폰: o
키뮤어 쿠폰: x
커버낫 쿠폰: x
무신사 스탠다드 쿠폰: x
라퍼지스토어 쿠폰: o
커버낫 쿠폰: x
라퍼지스토어 쿠폰: o
후아유 쿠폰: x
프리즘웍스 쿠폰: x
무신사 스탠다드 쿠폰: x
엠엘비 쿠폰: x
비슬로우 쿠폰: x
인사일런스 쿠폰: x
캘빈클라인 진 쿠폰: x
스파

In [71]:
# 리뷰 수 
points = soup.select('div.right_area.page-contents-best form#goodsRankForm div.list-box.box div.article_info p.point span.count')

for point in points :
    print(point.text)

4,749
11,419
657
52,489
222
2,657
17,285
7,932
3,901
5,562
5,898
660
520
12,257
952
9,639
378
19,046
3,062
189
1,131
4,317
8,217
5,822
9,067
5,044
443
38
357
6,946
27,405
4,593
197
2,174
339
235
11,558
2,498
6,458
6,554
1,952
1,621
1,092
1,892
457
609
519
513
3,417
624
12,533
7,292
186
261
10,443
275
1,828
200
12,344
6,071
120
1,505
77
351
488
80
1,247
289
838
379
181
1,594
46,157
36
7,521
1,568
508
147
24,328
54
950
101
81
7,888
8,954
10,555
160
47
134
5,944


In [42]:
ii = 0
while ii < 90 :

    
    # 브랜드
    brands = soup.select('form#goodsRankForm div.list-box.box div.article_info p.item_title a')
    
    brd = "'"+brands[ii].text+"'"
        
    # 제품명
    names = soup.select('form#goodsRankForm div.list-box.box div.article_info p.list_info a')
    
    nm = "'"+(names[ii]['title'])+"'"
        
    # 다 넣은 곳에서 쿠폰 있는거 없는거 찾기
    all_items = soup.select('div.right_area.page-contents-best form#goodsRankForm div.list-box.box .li_box')
    
    a = all_items[ii].find('p', { 'class' : 'mu-icon mu-icon-coupon'})
    if a == None :
            coup =  "'x'"
    else :
            coup =  "'o'"
            
    # 리뷰 수 
    points = soup.select('div.right_area.page-contents-best form#goodsRankForm div.list-box.box div.article_info p.point span.count')
    
    pp = points[ii].text.replace(',','')
    
    # 가격대
    prices = soup.select('div.right_area.page-contents-best form#goodsRankForm div.list-box.box div.article_info p.price')
        
    price11 = prices[ii].text.split()
        
    if len(price11) == 2:
        pri1 = price11[0][0:-1].replace(',','')
        pri2 = price11[1][0:-1].replace(',','')
    else:
        pri1 = price11[0][0:-1].replace(',','')
        pri2 = 0
    
    print('INSERT INTO musinsa VALUES (',ii+1, ',', brd, ',', nm, ',', pri1, ',', pri2, ',', pp , ',', coup ,');')
    
    ii += 1

INSERT INTO musinsa VALUES ( 1 , '드로우핏' , '프리미어 퓨어 캐시미어 히든 싱글 코트 [BLACK]' , 278000 , 219000 , 4752 , 'x' );
INSERT INTO musinsa VALUES ( 2 , '스파오' , '베이직 푸퍼_SPJPC4TC11' , 69900 , 0 , 11421 , 'o' );
INSERT INTO musinsa VALUES ( 3 , '내셔널지오그래픽' , 'N224MDW920 헤론 U넥 경량 다운 점퍼 CARBON BLACK' , 149000 , 0 , 657 , 'x' );
INSERT INTO musinsa VALUES ( 4 , '토피' , '2WAY 스웻 후드 집업 (MELANGE GREY)' , 45000 , 42000 , 52500 , 'x' );
INSERT INTO musinsa VALUES ( 5 , '노스페이스' , 'NJ1DN75A 남성 1996 에코 눕시 자켓' , 339000 , 0 , 222 , 'x' );
INSERT INTO musinsa VALUES ( 6 , '쿠어' , 'MTR 발마칸 코트 (블랙)' , 329000 , 296100 , 2658 , 'x' );
INSERT INTO musinsa VALUES ( 7 , '라퍼지스토어' , '[리뉴얼 ver.][3M 신슐레이트]오리지널 M-1965 피쉬테일 파카_Original Khaki' , 138000 , 99900 , 17288 , 'o' );
INSERT INTO musinsa VALUES ( 8 , '스파오' , '베이직 퍼플리스 집업_SPFZC4TU01' , 29900 , 0 , 7938 , 'o' );
INSERT INTO musinsa VALUES ( 9 , '파르티멘토' , '[22F/W 리뉴얼] 오버사이즈 멜톤 더플 롱 코트 네이비' , 155000 , 124000 , 3899 , 'o' );
INSERT INTO musinsa VALUES ( 10 , '스파오' , '파스텔 푸퍼_S

INSERT INTO musinsa VALUES ( 79 , '무신사 스탠다드' , '오버사이즈 블레이저 [블랙]' , 83900 , 75490 , 24328 , 'x' );
INSERT INTO musinsa VALUES ( 80 , '엠엘비' , '베이직 스몰로고 숏패딩 NY (Black)' , 289000 , 0 , 54 , 'x' );
INSERT INTO musinsa VALUES ( 81 , '비슬로우' , '22FW OVERSIZED M65 HOOD SHORT DOWN PARKA BLACK' , 269000 , 228650 , 950 , 'x' );
INSERT INTO musinsa VALUES ( 82 , '인사일런스' , 'MTR 캐시미어 발마칸 코트 BLACK' , 422000 , 337600 , 104 , 'x' );
INSERT INTO musinsa VALUES ( 83 , '캘빈클라인 진' , '여 아이보리 숏 렝스 스탠드 카라 푸퍼 다운 재킷 J221163 ACF' , 279000 , 251100 , 81 , 'x' );
INSERT INTO musinsa VALUES ( 84 , '스파오' , '리버서블 푸퍼_SPJDC4VC14' , 89900 , 80910 , 7888 , 'o' );
INSERT INTO musinsa VALUES ( 85 , '토피' , '2WAY 스웻 후드 집업 (BLACK)' , 45000 , 42000 , 8953 , 'x' );
INSERT INTO musinsa VALUES ( 86 , '파르티멘토' , '3M 신슐레이트 M-65 피시테일 코트 패딩 카키' , 155000 , 99900 , 10555 , 'o' );
INSERT INTO musinsa VALUES ( 87 , '쿠어' , '오버사이즈 클래식 발마칸 코트 (멜란지차콜)' , 372000 , 334800 , 161 , 'x' );
INSERT INTO musinsa VALUES ( 88 , '디스이즈네버댓' , 'PERTEX® T Dow

In [21]:
# 누적판매량 추가
ii = 0
driver = webdriver.Chrome()
while ii < 90 :

    driver.get("https://www.musinsa.com/ranking/best?period=month&age=ALL&mainCategory=002&subCategory=&leafCategory=&price=&golf=false&kids=false&newProduct=false&exclusive=false&discount=false&soldOut=false&page=1&viewType=small&priceMin=&priceMax=")
    items = driver.find_elements_by_css_selector("img.lazyload.lazy")
    items[ii].click()
    time.sleep(1)
    sum_sale = driver.find_element_by_id("sales_1y_qty")
    sst = sum_sale.text

    if '만' in sst :
        if '.' in sst : 
            sst = sst.replace('.','')
            sst += '000'
        else :
            sst += '0000'
            
        sst = sst.replace('만 개 이상','')
        
    elif '천' in sst:
        if '.' in sst : 
            sst = sst.replace('.','')
            sst += '00'
        else :
            sst += '000'
        sst = sst.replace('천 개 이상','')
        
    else :
        sst = sst.replace('개 이상','')
        
    
    # 브랜드
    brands = soup.select('form#goodsRankForm div.list-box.box div.article_info p.item_title a')
    
    brd = "'"+brands[ii].text+"'"
        
    # 제품명
    names = soup.select('form#goodsRankForm div.list-box.box div.article_info p.list_info a')
    
    nm = "'"+(names[ii]['title'])+"'"
        
    # 다 넣은 곳에서 쿠폰 있는거 없는거 찾기
    all_items = soup.select('div.right_area.page-contents-best form#goodsRankForm div.list-box.box .li_box')
    
    a = all_items[ii].find('p', { 'class' : 'mu-icon mu-icon-coupon'})
    if a == None :
            coup =  "'x'"
    else :
            coup =  "'o'"
            
    # 리뷰 수 
    points = soup.select('div.right_area.page-contents-best form#goodsRankForm div.list-box.box div.article_info p.point span.count')
    
    pp = points[ii].text.replace(',','')
    
    # 가격대
    prices = soup.select('div.right_area.page-contents-best form#goodsRankForm div.list-box.box div.article_info p.price')
        
    price11 = prices[ii].text.split()
        
    if len(price11) == 2:
        pri1 = price11[0][0:-1].replace(',','')
        pri2 = price11[1][0:-1].replace(',','')
    else:
        pri1 = price11[0][0:-1].replace(',','')
        pri2 = 0
    
    print('INSERT INTO musinsa VALUES (',ii+1, ',', brd, ',', nm, ',', pri1, ',', pri2, ',', pp , ',', coup , ',', sst, ');')
    
    ii += 1

INSERT INTO musinsa VALUES ( 1 , '드로우핏' , '프리미어 퓨어 캐시미어 히든 싱글 코트 [BLACK]' , 278000 , 208500 , 4805 , 'x' , 8600 );
INSERT INTO musinsa VALUES ( 2 , '스파오' , '베이직 푸퍼_SPJPC4TC11' , 69900 , 62910 , 11741 , 'x' , 41000 );
INSERT INTO musinsa VALUES ( 3 , '노스페이스' , 'NJ1DN75A 남성 1996 에코 눕시 자켓' , 339000 , 0 , 277 , 'x' , 2200 );
INSERT INTO musinsa VALUES ( 4 , '내셔널지오그래픽' , 'N224MDW920 헤론 U넥 경량 다운 점퍼 CARBON BLACK' , 149000 , 0 , 720 , 'x' , 5200 );
INSERT INTO musinsa VALUES ( 5 , '쿠어' , 'MTR 발마칸 코트 (블랙)' , 329000 , 296100 , 2732 , 'o' , 4700 );
INSERT INTO musinsa VALUES ( 6 , '토피' , '2WAY 스웻 후드 집업 (MELANGE GREY)' , 45000 , 37000 , 52900 , 'x' , 149000 );
INSERT INTO musinsa VALUES ( 7 , '라퍼지스토어' , '[리뉴얼 ver.][3M 신슐레이트]오리지널 M-1965 피쉬테일 파카_Original Khaki' , 138000 , 89900 , 17402 , 'o' , 37000 );
INSERT INTO musinsa VALUES ( 8 , '파르티멘토' , '[22F/W 리뉴얼] 오버사이즈 멜톤 더플 롱 코트 네이비' , 155000 , 99900 , 3990 , 'x' , 15000 );
INSERT INTO musinsa VALUES ( 9 , '스파오' , '베이직 퍼플리스 집업_SPFZC4TU01' , 29900 , 26910

INSERT INTO musinsa VALUES ( 74 , '캘빈클라인 진' , '여 아이보리 숏 렝스 스탠드 카라 푸퍼 다운 재킷 J221163 ACF' , 279000 , 251100 , 97 , 'x' , 600 );
INSERT INTO musinsa VALUES ( 75 , '스파오' , '리버서블 푸퍼_SPJDC4VC14' , 89900 , 80910 , 7921 , 'x' , 24000 );
INSERT INTO musinsa VALUES ( 76 , '키뮤어' , '서플러스 울(WOOL) 65 세미오버 더플 코트_네이비' , 209000 , 169000 , 217 , 'x' , 900 );
INSERT INTO musinsa VALUES ( 77 , '인사일런스' , '캐시미어 더블 롱 코트 BLACK' , 299000 , 164450 , 10769 , 'x' , 6000 );
INSERT INTO musinsa VALUES ( 78 , '디스이즈네버댓' , 'PERTEX® T Down Jacket Black' , 219000 , 153300 , 66 , 'x' , 900 );
INSERT INTO musinsa VALUES ( 79 , '후아유' , '리버서블 덤블 숏푸퍼 [5color] WHJPC4V02U' , 129000 , 89900 , 306 , 'x' , 1400 );
INSERT INTO musinsa VALUES ( 80 , '라퍼지스토어' , '[리뉴얼 ver.][3M 신슐레이트] M-1965 피쉬테일 숏 파카_Original Khaki' , 129000 , 79900 , 1620 , 'o' , 5700 );
INSERT INTO musinsa VALUES ( 81 , '캘빈클라인 진' , '여 블랙 숏 렝스 스탠드 카라 푸퍼 다운 재킷 J221163 BEH' , 279000 , 251100 , 71 , 'x' , 500 );
INSERT INTO musinsa VALUES ( 82 , '무신사 스탠다드' , '오버사이즈 블레이



<테이블 예시>
create table product (
            ranks int not null,
            brand varchar(20) not null,
            title varchar(150) not null,
            ori_price varchar(20),
            dis_price varchar(20),
            review_count int,
            coupon varchar(20)
            primary key(ranks)
);